cadc-storage-adapter-ad/src/main/java/org/opencadc/inventory/storage/ad/AdStorageQuery.java

```diff
@@ -134,14 +134,17 @@ public AdStorageMetadataRowMapper() { }
         public StorageMetadata mapRow(List<Object> row) {
             Iterator i = row.iterator();
 
-            String archive = (String) i.next();
-            String fname = (String) i.next();
             URI uri = (URI) i.next();
             if (uri == null) {
-                log.warn(AdStorageMetadataRowMapper.class.getSimpleName() + ".SKIP loc=" + archive + "/" + fname + " reason=null-uri");
+                log.warn(AdStorageMetadataRowMapper.class.getSimpleName() + ".SKIP loc=null/null reason=null-uri");
                 return null;
             }
             
+            // trust uri, <scheme>:<archive>/<fname>
+            String archiveFname = uri.toString().split(":")[1];
```

Comment:
The URI from line 137 is now the one that should be used directly as the storageID; it doesn't need to be deconstructed and reconstructed here. This applies to everything down to line ~153 where `URI storageID` is declared. 

The error message on line 139: the  "loc=null/null" is useless and should be removed; that warning message won't be very useful without a file-specific identifier but that's probably OK for now.

In the warning messages below, just use `"uri=" + uri` since that's never null.

Reply:
Updated AdStorageQuery.